### ISSUE TITLE
VisibilityNotifier2D can't resolve its connected signals when opened in the editor

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#5874]
**Operating system or device - Godot version:**
Mageia 6 x64, git master HEAD

**Issue description** (what happened, and what was expected):
When opening a scene in the editor with a VisibilityNotifier2D which has connected signals, errors are printed in the console mentioning that the corresponding callbacks can't be resolved, e.g.:
```
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Node2D(Node2D.gd)::_on_VisibilityNotifier2D_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Node2D(Node2D.gd)::_on_VisibilityNotifier2D_exit_screen': Method not found.
   At: core/object.cpp:1215.
```

These errors are misleading and wrong, as those methods exist in the referenced script.

**Steps to reproduce:**
1. Create a scene with a VisibilityNotified2D and a script
2. Bind the `_enter_screen` and `_exit_screen` signals to functions in the script.
3. Save the scene and close it.
4. Reopen the scene; the console should should errors as above.

**Link to minimal example project** (optional but very welcome):
[bug_signals_in_editor.zip](https://github.com/godotengine/godot/files/379843/bug_signals_in_editor.zip)

**Comment by akien-mga** [comment]
It can be very spammy:
```
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Node2D(lock.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Node2D(lock.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Node2D(lock.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Node2D(lock.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Node2D(lock.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Node2D(lock.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
Also saving modified external resource: res://tilesets/mountain_tileset.tres
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Node2D(lock.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Node2D(lock.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
Also saving modified external resource: res://tilesets/mountain_tileset.tres
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'StaticBody2D(house.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Node2D(lock.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
MUST RELOAD? 1
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(big_coin.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Node2D(lock.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(big_coin.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'StaticBody2D(house.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'enter_screen': 'Area2D(wise_alpaca.gd)::_on_enter_screen': Method not found.
   At: core/object.cpp:1215.
ERROR: emit_signal: Error calling method from signal 'exit_screen': 'Area2D(wise_alpaca.gd)::_on_exit_screen': Method not found.
   At: core/object.cpp:1215.
```
:)

**Comment by akien-mga** [comment]
Fixed by 37b5e99.
